autora/experimentalist/filter.py

```diff
@@ -1,2 +1,98 @@
+from enum import Enum
+
+import numpy as np
+
+
 def weber_filter(values):
     return filter(lambda s: s[0] <= s[1], values)
+
+
+def train_test_filter(seed=180, train_p=0.5):
+    """
+    A pipeline filter which pseudorandomly assigns values from the input into "train" or "test"
+    groups.
+
+    Examples:
+        We can create complementary train and test filters using the function:
+        >>> train_filter, test_filter = train_test_filter(train_p=0.6, seed=180)
+
+        The train filter generates a sequence of 60% of the input list.
+        >>> list(train_filter(range(20)))
+        [0, 2, 3, 4, 5, 6, 9, 10, 11, 12, 15, 16, 17, 18, 19]
```

Comment:
There is one thing, that we might consider: This function does not split the data into 60% train and 40% test data. Instead of that, each "new value" is put into train data with a probability of 60% and into test data with a probability of 40%. (For example, here we have 15 datapoints put into train and 5 into test, while a "perfect" split would be 12 into train and 8 into test). If there are a lot of datapoints the difference might not be huge, but if we have only little, we might need a different approach.

Reply:
Hi Younes, that's a really good point, and I've added some additional documentation to reflect that, in particular:
- a warning that this only works in the limit
- a suggestion that if you can enumerate the data in advance, you should use the scikit-learn `train_test_split` function instead.

Reply:
I think that's a good point. For the initial validation experiments, it would be better to use the train_test_split method since that ensures a clean cut. I think the documentation makes that very clear now.